src/Ocelot/WebSockets/Middleware/WebSocketsProxyMiddleware.cs

```diff
@@ -48,13 +48,21 @@ private static async Task PumpWebSocket(WebSocket source, WebSocket destination,
                 catch (OperationCanceledException)
                 {
                     await destination.CloseOutputAsync(WebSocketCloseStatus.EndpointUnavailable, null, cancellationToken);
+                    if (destination.State == WebSocketState.Open || destination.State == WebSocketState.CloseReceived)
+                    {
+                        await destination.CloseOutputAsync(WebSocketCloseStatus.EndpointUnavailable, null, cancellationToken);
+                    }
```

Comment:
Simple logical challenge:
How many times will the action be applied here?
Once or twice?

The action is `CloseOutputAsync`.